src/renderer/src/views/Home/Settings/ProjectConfig/MemberManagmentModal/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { Dialog } from '@mui/material'
+import { LeaveProjectContent } from './LeaveProjectContent'
+import { DeviceInfoContent } from './DeviceInfoContent'
+
+export type MemberManagementModalContent = { type: 'leaveProject' } | { type: 'memberInfo'; deviceId: string }
+
+type LeaveProjectModalProps = {
+  isOpen: boolean
+  closeModal: () => void
+  content: MemberManagementModalContent
+  resetOnDelete: () => void
+}
+
+export const MemberManagementModal = ({
+  isOpen,
+  closeModal,
+  content,
+  resetOnDelete,
+}: LeaveProjectModalProps) => {
+  function handleCloseDialog(event?: Record<string, never>, reason?: 'escapeKeyDown' | 'backdropClick') {
+    if (reason === 'backdropClick' || reason === 'escapeKeyDown') return
+    closeModal()
+  }
+
+  return (
+    <Dialog
+      transitionDuration={{ exit: 0 }}
```

Comment:
when a user gets deleted, the modal is closed. There is a slight flash between when the user gets deleted and when the modal closes if there is a non-zero transition duration as zustand is synchronous

Reply:
probably not worth trying to implement but if you're trying to avoid content flashes in the modal while it's closing, you can use the [`TransitionProps` prop on the Dialog](https://mui.com/material-ui/api/dialog/#props) to handle some cases. May be harder to coordinate for your case, but in case it's useful info.

example below where i use it to reset the modal state after exiting the success state:

https://github.com/digidem/mapeo-desktop-shell/blob/bb4ff4a4ad37b2e2217545da2d4d05d54d937abe/src/renderer/src/views/Home/Settings/ProjectConfig/InviteDeviceModal/Layout.tsx#L34-L36

https://github.com/digidem/mapeo-desktop-shell/blob/bb4ff4a4ad37b2e2217545da2d4d05d54d937abe/src/renderer/src/views/Home/Settings/ProjectConfig/InviteDeviceModal/index.tsx#L38-L47